diff --git a/app/Model/Attribute.php b/app/Model/Attribute.php
index e4269ce98..aefb76337 100644
--- a/app/Model/Attribute.php
+++ b/app/Model/Attribute.php
@@ -1576,7 +1576,7 @@ class Attribute extends AppModel
     {
         if ($jobId) {
             $message = $attributeId ? __('Correlating Attribute %s', $attributeId) : __('Correlating Event %s (%s MB used)', $eventId, intval(memory_get_usage() / 1024 / 1024));
-            $this->Job->saveProgress($jobId, $message, ($j / $eventCount) * 100);
+            $this->Job->saveProgress($jobId, $message, ($j == 0 or $eventCount == 0) ? 0 : ($j / $eventCount) * 100);
         }
         $attributeConditions = [
             'Attribute.deleted' => 0,